packages/@aws-cdk/core-synthesizer/lib/staging-stack.ts

```diff
@@ -0,0 +1,184 @@
+
+  /**
+   * The app-scoped, environment-keyed repositories created in this staging stack.
+   * A repository is created per image asset family.
+   */
+  readonly stagingRepos: Record<string, ecr.Repository>;
+
+  /**
+   * // TODO
+   */
+  addFile(asset: FileAssetSource): FileAssetInfo;
+
+  /**
+   * // TODO
+   */
+  addDockerImage(asset: DockerImageAssetSource): DockerAssetInfo;
+}
+
+/**
+ * Staging Stack Properties
+ */
+export interface StagingStackProps extends StackProps {
+  /**
+   * Explicit name for the staging bucket
+   *
+   * @default - a well-known name unique to this app/env.
+   */
+  readonly stagingBucketName?: string;
+
+  /**
+   * Pass in an existing role to be used as the file publishing role.
+   *
+   * @default - a well-known name unique to this app/env.
+   */
+  readonly fileAssetPublishingRoleName?: string;
+
+  /**
+   * Pass in an existing role to be used as the image publishing role.
+   *
+   * @default - a well-known name unique to this app/env.
+   */
+  readonly dockerAssetPublishingRoleName?: string;
+}
+
+/**
+ * A default Staging Stack
+ */
+export class DefaultStagingStack extends Stack implements IStagingStack {
+  /**
+   * Default asset publishing role name for file (S3) assets.
+   */
+  private static readonly DEFAULT_FILE_ASSET_PUBLISHING_ROLE_NAME = 'cdk-${Qualifier}-file-publishing-role-${AWS::AccountId}-${AWS::Region}';
+
+  /**
+   * Default asset publishing role name for docker (ECR) assets.
+   */
+  private static readonly DEFAULT_DOCKER_ASSET_PUBISHING_ROLE_NAME = 'cdk-${Qualifier}-asset-publishing-role-${AWS::AccountId}-${AWS::Region}';
+
+  /**
+   * The app-scoped, evironment-keyed staging bucket.
+   */
+  public readonly stagingBucket?: s3.Bucket;
+
+  /**
+   * The app-scoped, environment-keyed ecr repositories associated with this app.
+   */
+  public readonly stagingRepos: Record<string, ecr.Repository>;
+
+  private readonly stagingBucketName?: string;
+  private fileAssetPublishingRoleName: string;
+  private dockerAssetPublishingRoleName: string;
+
+  constructor(scope: Construct, id: string, props: StagingStackProps = {}) {
+    super(scope, id, props);
+
+    this.stagingBucketName = props.stagingBucketName;
+    this.fileAssetPublishingRoleName = props.fileAssetPublishingRoleName ?? DefaultStagingStack.DEFAULT_FILE_ASSET_PUBLISHING_ROLE_NAME;
+    this.dockerAssetPublishingRoleName = props.dockerAssetPublishingRoleName ?? DefaultStagingStack.DEFAULT_DOCKER_ASSET_PUBISHING_ROLE_NAME;
+    this.stagingRepos = {};
+  }
+
+  /**
+   * Returns the well-known name of the file publishing role
+   */
+  private getCreateFilePublishingRole() {
+    this.node.tryFindChild(this.fileAssetPublishingRoleName) as iam.Role ?? new iam.Role(this, this.fileAssetPublishingRoleName, {
+      roleName: DefaultStagingStack.DEFAULT_FILE_ASSET_PUBLISHING_ROLE_NAME,
+      assumedBy: new iam.ServicePrincipal('sts.amazonaws.com'), // TODO actually create correct role
```

Comment:
Check the current assume role policies. I think the correct assumer here will be:

- The Deploy Role if the user chooses a Deploy Role (should be a parameter to the stack synthesizer)
- The current account root if not

Reply:
Perhaps this needs to be configurable?